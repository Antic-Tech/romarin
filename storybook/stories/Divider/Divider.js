import React from 'react';
import { StyleSheet, View } from 'react-native';
import { Text } from '../../components';

import ThemeProvider from '../../theme';
import { PixelRatio } from 'react-native';

const getDividerStyle = () => ({
	backgroundColor: ThemeProvider.theme.dividerColor,
	height: 1 / PixelRatio.get(),
	flex: 1
});

export default ({ label, ...rest }) =>
	label ? (
		<View
			style={{
				paddingBottom: ThemeProvider.theme.paddingMedium,
				flexDirection: 'row',
				flex: 1,
				paddingVertical: ThemeProvider.theme.paddingXSmall,

				alignItems: 'center'
			}}
			{...rest}
		>
			<View style={getDividerStyle()} />
			<Text
				style={{
					paddingLeft: ThemeProvider.theme.dividerLabelPadding,
					paddingRight: ThemeProvider.theme.dividerLabelPadding
				}}
				size="small"
				bold
			>
				{label}
			</Text>
			<View style={getDividerStyle()} />
		</View>
	) : (
		<View
			style={{
				flexDirection: 'row',
				flex: 1,
				alignItems: 'center'
			}}
			{...rest}
		>
			<View style={getDividerStyle()} />
		</View>
	);
