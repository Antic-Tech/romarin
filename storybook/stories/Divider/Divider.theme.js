export default core => ({
	dividerColor: core.colorDarkerGray,
	dividerLabelPadding: core.corePaddingSmall
});
