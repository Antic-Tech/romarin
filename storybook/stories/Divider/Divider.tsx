import React from 'react';
import { PixelRatio, View } from 'react-native';
import { Text } from '../../components';
import { withTheme } from '../../theme';

export default withTheme(
	({ label, theme, ...rest }) =>
		label ? (
			<View
				style={{
					paddingBottom: theme.paddingMedium,
					flexDirection: 'row',
					flex: 1,
					paddingVertical: theme.paddingXSmall,

					alignItems: 'center'
				}}
				{...rest}
			>
				<View
					style={{
						backgroundColor: theme.dividerColor,
						height: 1 / PixelRatio.get(),
						flex: 1
					}}
				/>
				<Text
					style={{
						paddingLeft: theme.dividerLabelPadding,
						paddingRight: theme.dividerLabelPadding
					}}
					size="small"
					bold
				>
					{label}
				</Text>
				<View
					style={{
						backgroundColor: theme.dividerColor,
						height: 1 / PixelRatio.get(),
						flex: 1
					}}
				/>
			</View>
		) : (
			<View
				style={{
					flexDirection: 'row',
					flex: 1,
					alignItems: 'center'
				}}
				{...rest}
			>
				<View
					style={{
						backgroundColor: theme.dividerColor,
						height: 1 / PixelRatio.get(),
						flex: 1
					}}
				/>
			</View>
		)
);
