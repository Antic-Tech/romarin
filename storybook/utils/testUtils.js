import { mount, shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

export const renderComponent = (jsx, { shallowRender = true, store } = {}) => {
	const renderOptions = { context: { store } };

	if (shallowRender) {
		return shallow(jsx, renderOptions);
	}

	return mount(jsx, renderOptions);
};

export const componentFactory = Component => (props = {}, options = {}) => {
	return renderComponent(<Component {...props} />, options);
};

export const getTree = jsx => renderer.create(jsx).toJSON();

export const verifySnapshot = jsx => {
	const tree = getTree(jsx);

	expect(tree).toMatchSnapshot();
};
