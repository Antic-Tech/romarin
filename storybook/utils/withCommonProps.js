import React from 'react';
import { StyleSheet } from 'react-native';
import ThemeProvider from '../theme';

const Sizes = {
	XXSmall: 'XXSmall',
	XSmall: 'XSmall',
	Small: 'Small',
	Medium: 'Medium',
	Large: 'Large',
	XLarge: 'XLarge',
	XXLarge: 'XXLarge'
};

const withPadding = (Component) =>
	class WithLoading extends React.Component {
		render() {
			const {
				style,
				padding,
				paddingHorizontal,
				paddingVertical,
				paddingTop,
				paddingBottom,
				paddingLeft,
				paddingRight,
				...props
			} = this.props;

			const newStyles = StyleSheet.flatten([
				{
					padding: padding ? ThemeProvider.theme[`corePadding${padding}`] : undefined,
					paddingHorizontal: paddingHorizontal
						? ThemeProvider.theme[`corePadding${paddingHorizontal}`]
						: undefined,
					paddingVertical: paddingVertical ? ThemeProvider.theme[`corePadding${paddingVertical}`] : undefined,
					paddingTop: paddingTop ? ThemeProvider.theme[`corePadding${paddingTop}`] : undefined,
					paddingBottom: paddingBottom ? ThemeProvider.theme[`corePadding${paddingBottom}`] : undefined,
					paddingLeft: paddingLeft ? ThemeProvider.theme[`corePadding${paddingLeft}`] : undefined,
					paddingRight: paddingRight ? ThemeProvider.theme[`corePadding${paddingRight}`] : undefined
				},
				style
			]);

			return <Component style={newStyles} {...props} />;
		}
	};

const withMargin = (Component) =>
	class WithLoading extends React.Component {
		render() {
			const {
				style,
				margin,
				marginHorizontal,
				marginVertical,
				marginTop,
				marginBottom,
				marginLeft,
				marginRight,
				...props
			} = this.props;

			const newStyles = StyleSheet.flatten([
				{
					margin: margin ? ThemeProvider.theme[`coreMargin${margin}`] : undefined,
					marginHorizontal: marginHorizontal
						? ThemeProvider.theme[`coreMargin${marginHorizontal}`]
						: undefined,
					marginVertical: marginVertical ? ThemeProvider.theme[`coreMargin${marginVertical}`] : undefined,
					marginTop: marginTop ? ThemeProvider.theme[`coreMargin${marginTop}`] : undefined,
					marginBottom: marginBottom ? ThemeProvider.theme[`coreMargin${marginBottom}`] : undefined,
					marginLeft: marginLeft ? ThemeProvider.theme[`coreMargin${marginLeft}`] : undefined,
					marginRight: marginRight ? ThemeProvider.theme[`coreMargin${marginRight}`] : undefined
				},
				style
			]);

			return <Component style={newStyles} {...props} />;
		}
	};

const withCommonProps = (Component) => withPadding(withMargin(Component));

export { withPadding, withMargin, withCommonProps };
