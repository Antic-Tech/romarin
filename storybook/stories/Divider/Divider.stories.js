import React from 'react';

import { storiesOf } from '@storybook/react-native';
import { action } from '@storybook/addon-actions';

import Divider from './Divider';
import { Text } from '../../components';
import CenterView from '../CenterView';

storiesOf('Divider', module)
	.addDecorator(getStory => <CenterView>{getStory()}</CenterView>)
	.add('default', () => (
		<React.Fragment>
			<Divider />
			<Divider label="Label" />
		</React.Fragment>
	));
