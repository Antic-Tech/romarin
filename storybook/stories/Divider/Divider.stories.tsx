import { storiesOf } from '@storybook/react-native';
import React from 'react';
import { Page } from '../../components';
import Divider from './Divider';



storiesOf('Divider', module)
	.addDecorator(getStory => <Page>{getStory()}</Page>)
	.add('default', () => (
		<React.Fragment>
			<Divider />
			<Divider label="Label" />
		</React.Fragment>
	));
