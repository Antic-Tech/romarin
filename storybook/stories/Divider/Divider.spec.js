import React from 'react';
import Divider from './Divider';

import { verifySnapshot } from '../../utils/testUtils';

describe('Divider', () => {
	describe('Snapshots', () => {
		it('renders correctly', () => {
			verifySnapshot(<Divider />);
		});

		it('renders correctly with label', () => {
			verifySnapshot(<Divider label="Label" />);
		});
	});
});
