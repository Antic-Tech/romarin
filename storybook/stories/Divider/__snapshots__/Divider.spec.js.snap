// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Snapshots renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#181818",
        "flex": 1,
        "height": 0.5,
      }
    }
  />
</View>
`;

exports[`Divider Snapshots renders correctly with label 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "paddingBottom": undefined,
      "paddingVertical": undefined,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#181818",
        "flex": 1,
        "height": 0.5,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": undefined,
          "fontSize": 13,
        },
        Object {
          "paddingLeft": 8,
          "paddingRight": 8,
        },
      ]
    }
  >
    Label
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#181818",
        "flex": 1,
        "height": 0.5,
      }
    }
  />
</View>
`;
